components/tidb_query/src/batch/runner.rs

```diff
@@ -304,6 +310,11 @@ impl<SS: 'static> BatchExecutorsRunner<SS> {
         let executors_len = req.get_executors().len();
         let collect_exec_summary = req.get_collect_execution_summaries();
         let config = Arc::new(EvalConfig::from_request(&req)?);
+        let encode_type = if req.has_encode_type() {
+            req.get_encode_type()
+        } else {
+            EncodeType::TypeDefault
```

Comment:
TypeDefault should be the default type and maybe you don't need to use `if req.has_encode_type()`